src/tasks/RuntimeconfigParser/RuntimeconfigParser.cs

```diff
@@ -0,0 +1,108 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Reflection.Metadata;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+
+public class RuntimeconfigParserTask : Task
+{
+    /// <summary>
+    /// The path to runtimeconfig.json file.
+    /// </summary>
+    [Required]
+    public string runtimeConfigFile { get; set; } = ""!;
+
+    /// <summary>
+    /// The path to the output binary file.
+    /// </summary>
+    [Required]
+    public string outputFile { get; set; } = ""!;
+
+    /// <summary>
+    /// List of reserved properties.
```

Comment:
What does it mean for the property to be reserved?

Reply:
These properties are supposed to be set by the mobile targets.

Reply:
We have to check for potential conflicts between the properties set in the runtimeconfig file and properties set directly by the host (in this case the Xamarin products). We want to do so at build time here to avoid slowing down startup, so the host passes a list of properties it reserves for that purpose.